llvm/lib/Analysis/InstructionSimplify.cpp

```diff
@@ -440,7 +440,8 @@ static Value *threadBinOpOverSelect(Instruction::BinaryOps Opcode, Value *LHS,
     // Check that the simplified value has the form "X op Y" where "op" is the
     // same as the original operation.
     Instruction *Simplified = dyn_cast<Instruction>(FV ? FV : TV);
-    if (Simplified && Simplified->getOpcode() == unsigned(Opcode)) {
+    if (Simplified && Simplified->getOpcode() == unsigned(Opcode) &&
+        !Simplified->hasPoisonGeneratingFlags()) {
```

Comment:
Shouldn't `isSafeToSpeculativelyExecute` suffice here?

Reply:
`isSafeToSpeculativelyExecute` only cares about immediate UB instead of poison.

Reply:
why is the fix to prohibit any poison flag and not to just drop the poison generating flag?

Reply:
IIRC modifying an instruction is not allowed in InstSimplify :(

Reply:
oh yeah xD

Maybe just move this to InstCombine? Or put equivelent code that drops poison generating flags there?